hogben/utils.py

```diff
@@ -130,31 +134,40 @@ def __corner(self, results):
         axes = np.reshape(np.array(fig.get_axes()), (self.ndim, self.ndim))
         for i in range(1, self.ndim):
             for j in range(self.ndim):
-                if i == self.ndim-1:
-                    axes[i,j].set_xlabel(self.params[j].name)
+                if i == self.ndim - 1:
+                    axes[i, j].set_xlabel(self.params[j].name)
                 if j == 0:
-                    axes[i,j].set_ylabel(self.params[i].name)
+                    axes[i, j].set_ylabel(self.params[i].name)
 
-        axes[self.ndim-1, self.ndim-1].set_xlabel(self.params[-1].name)
+        axes[self.ndim - 1, self.ndim - 1].set_xlabel(self.params[-1].name)
         return fig
 
-def fisher(qs, xi, counts, models, step=0.005):
+
+def fisher(qs: list[list],
+           xi: list[Union['refnx.analysis.Parameter',
+                          'bumps.parameter.Parameter']],
+           counts: list[int],
+           models: list[Union['refnx.reflect.ReflectModel',
+                              'refl1d.experiment.Experiment']],
+           step: float = 0.005,
+           importance: Optional[list[float]] = None) -> np.ndarray:
     """Calculates the Fisher information matrix for multiple `models`
        containing parameters `xi`.
 
     Args:
-        qs (list): Q points for each model.
-        xi (list): varying model parameters.
-        counts (list): incident neutron counts corresponding to each Q value.
-        models (list): models to calculate gradients with.
-        step (float): step size to take when calculating gradient.
+        qs: The Q points for each model.
+        xi: The varying model parameters.
+        counts: incident neutron counts corresponding to each Q value.
+        models: models to calculate gradients with.
+        step: step size to take when calculating gradient.
+        importance: The importance scaling for each parameter in xi
```

Comment:
just noticed that this is no longer correct as it isn't for each parameter in `xi`, we are using the `has attr` schema, so make it clear how this can be changed

Reply:
It actually doesn't use the argument at all anymore. So removed the entire line (and argument).

Reply:
Do change the docstring to make it clear how one could change the importance then (an example might be handy)